Rust formations (#760)
kgateway-dev/kgateway.dev: content/docs/envoy/latest/reference/release-notes.md

AI policy configuration was removed from the `TrafficPolicy` API. To use AI capabilities, use an [agentgateway proxy](https://agentgateway.dev/docs/kubernetes/latest/setup/) with the `AgentgatewayPolicy` API instead.

#### Default transformation engine changed to rustformation {#v22-default-transformation-engine}

> The default transformation engine flipped from the C++ classic filter (Inja) to rustformation (MiniJinja). See the [Rustformation transformation engine](#v22-rustformation) section under New features for the full set of capabilities. Any `TrafficPolicy` that has a `transformation` field is now processed by rustformation unless you explicitly opt out.

Reviewer: ```suggestion
The default transformation engine changed from the C++ classic filter (Inja) to rustformation (MiniJinja). For the full set of capabilities, see the [Rustformation transformation engine](#v22-rustformation) section under New features. Any `TrafficPolicy` that has a `transformation` field is now processed by rustformation unless you explicitly opt out.
```

The two engines are syntactically similar but not identical. Existing classic-engine templates from 2.1.x or earlier might render differently or fail to render under rustformation. Common patterns to watch for include:

* The `.0` accessor on JSON header fields (for example, `{{ headers.X-Incoming-Stuff.0 }}`). Rustformation requires bracket notation: `{{ headers["X-Incoming-Stuff"][0] }}`.
* Templates that rely on the body being auto-parsed as JSON. The classic engine auto-parses whenever a transformation is configured; rustformation defaults to `AsString`. Set `body.parseAs: AsJson` explicitly to keep dot-notation access working.
* Templates that use a JSON field name that collides with a built-in template function (such as `context`). With `parseAs: AsJson`, MiniJinja shadows the function with the field value and fails to render.

> * Whitespace handling: rustformation right-trims trailing whitespace by default; classic preserves whitespace as-is.

Reviewer: ```suggestion
* Whitespace handling. Rustformation right-trims trailing whitespace by default; classic preserves whitespace as-is.
```

> * The `replace_with_random` function is evaluated at template compile time when its arguments are constants and returns the same generated value on every request. To get a fresh random per request, pass a per-request input such as `request_header("x-request-id")`. Tracked in [kgateway-dev/kgateway#13634](https://github.com/kgateway-dev/kgateway/issues/13634).

Reviewer: idk if this is the best reword, but something about the first sentence is off

also I dont understand this bullet point in general - is this one of the patterns to watch out for or is this a bug?

```suggestion
* When the `replace_with_random` function's arguments are constants, it is evaluated at template compile time and returns the same generated value on every request. To get a fresh random per request, pass a per-request input such as `request_header("x-request-id")`. Tracked in [kgateway-dev/kgateway#13634](https://github.com/kgateway-dev/kgateway/issues/13634).
```

Author: I think it's a bug, but I removed the callout from here. The example used in the guide doesn't even require adjustment because of this bug. Better off just leave it one time in the templating-language topic just in case, but remove it from here.
